process/sync/shardForkDetector.go

```diff
@@ -40,6 +49,7 @@ func (sfd *shardForkDetector) AddHeader(
 	state process.BlockHeaderState,
 	finalHeaders []data.HeaderHandler,
 	finalHeadersHashes [][]byte,
+	isNotarizedShardStuck bool,
```

Comment:
this should be consumed from an interface, not given as parameter.

Reply:
Yep, I will think about it